socket_linux.go

```diff
@@ -26,13 +31,26 @@ func NewSocket(addr *net.UDPAddr, recvBuf int, reuseport bool) (net.PacketConn,
 		return nil, err
 	}
 
-	sockaddr := unix.SockaddrInet4{
-		Port: addr.Port,
-	}
-	if copied := copy(sockaddr.Addr[:], addr.IP); copied != net.IPv4len {
-		panic("did not copy enough bytes of ip address")
+	var sa unix.Sockaddr
+	if domain == unix.AF_INET {
+		sockaddr := &unix.SockaddrInet4{
+			Port: addr.Port,
+		}
+		if copied := copy(sockaddr.Addr[:], addr.IP); copied != net.IPv4len {
```

Comment:
Rather than using `copy` I think we can use `addr.IP.To4()` here and panic if for some reason it comes back nil?